### PR TITLE
Fix oss-fuzz 40243 [13161]

### PR DESCRIPTION
**Pull request by MiguelCompany**
Fixed a null-dereference on XML parser when a type has no `name` attribute

### CODE REVIEW

File: src/cpp/rtps/xmlparser/XMLDynamicParser.cpp [hunk 1]
```diff
@@ -606,6 +606,10 @@ XMLP_ret XMLParser::parseXMLBitmaskDynamicType(
     }
 
     const char* name = p_root->Attribute(NAME);
+    if (!name)
```
Comment (jsantiago-eProsima) [inline src/cpp/rtps/xmlparser/XMLDynamicParser.cpp:609]: I would test against nullptr just for consistency with the other test in this code block.

**Review by jsantiago-eProsima — CHANGES_REQUESTED** [review]
Small nitpick

File: src/cpp/rtps/xmlparser/XMLDynamicParser.cpp (hunk 1 above)
Comment (MiguelCompany) [reply src/cpp/rtps/xmlparser/XMLDynamicParser.cpp]: Changed

**Review by jsantiago-eProsima — APPROVED** [review]
LGTM

**Comment by MiguelCompany** [comment]
@mergifyio backport 2.4.x 2.3.x

**Pull request merged**